### PR TITLE
fix: initialize asar support in worker threads

### PR DESCRIPTION
**Pull request by indutny**
#### Description of Change

Use `ObjectWrap` instead of gin's Wrap in `electron_api_asar.cc` because
gin isn't fully initialized (and apparently not possible to initialize
without ruining the isolate configuration and array buffer allocator) in
worker threads. In the worker thread call `setupAsarSupport` just as we
do for the main process.

Test case: github.com/indutny-signal/asar-worker

cc @codebytere 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: fix: initialize asar support in worker threads

### CODE REVIEW

File: lib/asar/fs-wrapper.ts
Comment (RaisinTen) [inline lib/asar/fs-wrapper.ts:37]:
```suggestion
  } catch {
```
removing the `_` should be fine if we want to swallow the error. It is already done in places like https://github.com/electron/electron/blob/e589e9b259ecad5a2f2f47f0b8d40618d1c54601/lib/browser/api/app.ts#L77.

File: patches/node/worker_thread_add_asar_support.patch
Comment (RaisinTen) [inline patches/node/worker_thread_add_asar_support.patch:2]:
```suggestion
From: Fedor Indutny <indutny@signal.org>
```
extra `y`? :)

**Review by RaisinTen — COMMENTED** [review]
Would it make sense to add the test to `spec/asar-spec.js` and include the asar in `spec/fixtures`?

File: patches/node/worker_thread_add_asar_support.patch
Comment (indutny) [reply patches/node/worker_thread_add_asar_support.patch:2]: Well, it is a story for another day I guess.

**Comment by indutny-signal** [comment]
@RaisinTen good point. Added a test into `spec-main`'s asar test suite.

**Comment by indutny-signal** [comment]
I suppose I can also cc @zcbenz , not sure who the code owner is for this part of the code.

**Comment by indutny-signal** [comment]
(Friendly ping to keep the notifications up in inbox)

**Review by zcbenz — APPROVED** [review]

**Comment by indutny-signal** [comment]
Awesome. Thanks for approving! Do we need one more approval to merge it @zcbenz ?

**Comment by zcbenz** [comment]
I prefer waiting for one more approval, but if there is none I think it is ok to just merge this.

**Comment by indutny** [comment]
cc @MarshallOfSound @nornagon and just in case @codebytere

**Review by jkleinsc — APPROVED** [review]

**Comment by indutny-signal** [comment]
Thank you!

**Pull request merged**
